java/src/org/openqa/selenium/support/decorators/WebDriverDecorator.java

```diff
@@ -183,6 +189,60 @@
 @Beta
 public class WebDriverDecorator<T extends WebDriver> {
 
+  protected static class Definition {
+    private final Class<?> decoratedClass;
+    private final Class<?> originalClass;
+
+    public Definition(Decorated<?> decorated) {
+      this.decoratedClass = decorated.getClass();
+      this.originalClass = decorated.getOriginal().getClass();
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Did you ever consider using lombok? it greatly reduces the amount of boilrplate code

Reply:
This is a special equals implementation, it does perform a identity check. The Map of the cache will call this to lookup the key. So i don't think lombok will help here.